incubator-tools/best-practices/utilities/utilities.py

```diff
@@ -0,0 +1,743 @@
+    try:
+        bucket = storage_client.get_bucket(bucket_name)
+        bucket.delete(force=True)
+    except Exception:
+        pass
+
+
+def list_blobs(bucket_name):
+    """
+    Retrieves a list of filenames (blobs) from a specified Google Cloud Storage bucket.
+
+    Args:
+        bucket_name (str): The name of the bucket from which to retrieve the list of filenames.
+
+    Returns:
+        list: A list containing the names of all files (blobs) present in the specified bucket.
+    """
+
+    blob_list = []
+    storage_client = storage.Client()
+    blobs = storage_client.list_blobs(bucket_name)
+
+    for blob in blobs:
+        blob_list.append(blob.name)
+
+    return blob_list
+
+
+def matching_files_two_buckets(bucket_1, bucket_2):
+    """
+    Compares the files from two Google Cloud Storage buckets to find files with similar names.
+
+    Args:
+        bucket_1 (str): Name of the first GCS bucket.
+        bucket_2 (str): Name of the second GCS bucket.
+
+    Returns:
+        tuple: A tuple containing two dictionaries:
+            1. matched_files_dict: Dictionary with filenames from
+            bucket_1 as keys and corresponding similar filenames
+            from bucket_2 as values.
+            2. non_matched_files_dict: Dictionary with filenames from
+            bucket_1 as keys and a message indicating no similar file
+            was found in bucket_2 as values.
+    """
+
+    bucket_1_blobs = list_blobs(bucket_1)
+    bucket_2_blobs = list_blobs(bucket_2)
+
+    matched_files_dict = {}
+    non_matched_files_dict = {}
+
+    for i in bucket_1_blobs:
+        for j in bucket_2_blobs:
+            matched_score = difflib.SequenceMatcher(None, i, j).ratio()
+
+            print("matched_score:", matched_score)
+            if matched_score >= 0.8:
+                matched_files_dict[i] = j
+            else:
+                non_matched_files_dict[i] = "No parsed output available"
+
+    for i in matched_files_dict:
+        if i in non_matched_files_dict.keys():
+            del non_matched_files_dict[i]
+
+    print("matched_files_dict =", matched_files_dict)
+    print("non_matched_files_dict =", non_matched_files_dict)
+
+    return matched_files_dict, non_matched_files_dict
+
+
+def documentai_json_proto_downloader(bucket_name, blob_name_with_prefix_path):
+    """
+    Downloads a file from a specified Google Cloud Storage bucket and converts it into a DocumentAI Document proto.
+
+    Args:
+        bucket_name (str): The name of the GCS bucket from which to download the file.
+        blob_name_with_prefix_path (str): The full path (prefix) to the JSON blob in the bucket.
+
+    Returns:
+        documentai.Document: A DocumentAI Document proto representation of the downloaded JSON.
+    """
+
+    storage_client = storage.Client()
+    bucket = storage_client.bucket(bucket_name)
+    blob = bucket.blob(blob_name_with_prefix_path)
+
+    contents = blob.download_as_string()
```

Comment:
I think this should be `blob.download_as_bytes()` then dont use the `.decode()` method.